app/javascript/Pages/Home.jsx

```diff
@@ -1,37 +1,272 @@
-        <Head title="Welcome" />
-        <p>Hello, welcome to your first Inertia app!</p>
-        <p className="paragraph">This is cool!</p>
-
-        <Link
-          href="/logout"
-          method="delete"
-          headers={window.defaultHeaders}
-          as="button"
+import { Head } from "@inertiajs/inertia-react";
+import { useState, useEffect } from "react";
+import SocialCard from "./components/SocialCard";
+import "../stylesheets/home.scss";
+import ButtonDifficultyFilter from "./components/ButtonDifficultyFilter";
+import CardCategory from "./components/CardCategory";
+import CardPopular from "./components/CardPopular";
+import axios from "axios";
+import ButtonFilter from "./components/ButtonFilter";
+import PropTypes from "prop-types";
+import { useTranslation } from "react-i18next";
+
+function Homepage(props) {
+  const { t, i18n } = useTranslation();
+  const [items, setItems] = useState([]);
+  const [cats, setCat] = useState([]);
+  const [pclasses, setPClasses] = useState([]);
+  const [categories, setCategories] = useState([]);
+  const [categoryFilter, setCategoryFilter] = useState("");
+
+  const fetchClassesDifficulty = async (difficulty) => {
+    let newClassData = [];
+    axios
+      .get(`/api/classes?difficulty=${difficulty}`)
+      .then((response) => {
+        newClassData = response.data;
+        setItems(newClassData);
+      })
+      .catch((error) => {
+        console.log(error);
+        newClassData = [];
+      });
+  };
+
+  const fetchCategories = async () => {
+    let categoryData = [];
+    axios
+      .get("/api/skills")
+      .then((response) => {
+        categoryData = response.data;
+        setCat(categoryData);
+      })
+      .catch((error) => {
+        console.log(error);
+        categoryData = [];
+      });
+  };
+
+  const fetchPopularClasses = async () => {
+    let popularClassData = [];
+    axios
+      .get("/api/classes")
+      .then((response) => {
+        popularClassData = response.data;
+        setPClasses(popularClassData);
+      })
+      .catch((error) => {
+        console.log(error);
+        popularClassData = [];
+      });
+  };
+
+  const addCategoryButtons = async () => {
+    let categories = [];
+    axios
+      .get(`/api/skills`)
+      .then((response) => {
+        categories = response.data;
+
+        // const categoryComponents = [];
+
+        // categories.map((c, i) => {
+        //   categoryComponents.push(
+        //     <ButtonFilter
+        //       props={{
+        //         onClick: updateCategory,
+        //         params: c,
+        //         id: categoryFilter,
+        //       }}
+        //     >
+        //       {c.name}
+        //     </ButtonFilter>
+        //   );
+        // });
+
+        setCategories(categories);
+      })
+      .catch((error) => {
+        console.log(error);
+        categories = [];
+      });
+  };
+
+  const fetchClasses = async () => {
+    const params = new URLSearchParams();
```

Comment:
**suggestion (blocking)**: you don't need to do this for the params, you can pass a "params" object to axios (see the docs https://axios-http.com/docs/req_config)